### PR TITLE
Cleanup SGX entry code

### PR DESCRIPTION
**Pull request by jethrogb**
cc @aandyl

### CODE REVIEW

**Comment by nagisa** [comment]
All of these commits could definitely use some more text on them describing the why.

File: src/libstd/sys/sgx/abi/entry.S [hunk 1]
```diff
@@ -30,6 +30,14 @@ IMAGE_BASE:
 
 /*  We can store a bunch of data in the gap between MXCSR and the XSAVE header */
 
+/* MXCSR initialization value for ABI */
+.Lmxcsr_init:
+    .int 0x1f80
+
+/* x87 FPU control word initialization value for ABI */
+.Lfpucw_init:
+    .int 0x037f
```
Comment (nagisa) [inline src/libstd/sys/sgx/abi/entry.S]: This is setting one of the reserved bits (6th) to 1?

**Comment by alexcrichton** [comment]
Sorry I'm not on the libs team any more and I'd rather not be on the hook for these sorts of PRs. @nagisa mind taking over review?

File: src/libstd/sys/sgx/abi/entry.S (hunk 1 above)
Comment (jethrogb) [reply src/libstd/sys/sgx/abi/entry.S]: This is the default from the [Intel SDM, Vol. 1, §8.1.5](https://software.intel.com/sites/default/files/managed/39/c5/325462-sdm-vol-1-2abcd-3abcd.pdf)

**Comment by jethrogb** [comment]
See §4 of https://people.cs.kuleuven.be/~jo.vanbulck/ccs19-tale.pdf for the “why”.

**Comment by jethrogb** [comment]
This includes fixes for [INTEL-SA-00336](https://www.intel.com/content/www/us/en/security-center/advisory/intel-sa-00336.html)/[CVE-2020-0561](http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-0561), could this be merged soon?

**Comment by jethrogb** [comment]
Why does [bors](https://buildbot2.rust-lang.org/homu/queue/rust) show this as not mergeable?

**Pull request merged**